src/wallet/interfaces.cpp

```diff
@@ -643,6 +643,10 @@ class WalletLoaderImpl : public WalletLoader
         DatabaseStatus status;
         bilingual_str error;
         auto db = MakeWalletDatabase(wallet_name, options, status, error);
+        if (!db && status == wallet::DatabaseStatus::FAILED_LEGACY_DISABLED) {
+            options.require_format = wallet::DatabaseFormat::BERKELEY_RO;
+            db = MakeWalletDatabase(wallet_name, options, status, error);
+        }
```

Comment:
While adding this `BERKELEY_RO` database format fallback in this seemingly generic `isEncrypted` function looks unsafe, it is fine because the only usage of this function lies in the wallet migration functionality in the GUI. The `FAILED_LEGACY_DISABLED` error from `MakeDatabase` also enforces the flow towards wallet migration. 

Using the `require_format = DatabaseFormat::BERKELEY_RO` checks seems like an indirect way of validating the migration flow though when the `MakeWalletDatabase` function already decides to pick `BERKELEY_RO` format for BDB wallet files, perhaps a boolean might replace it in the future.